src/screens/Voucher/VoucherDetailScreen.tsx

```diff
@@ -63,7 +56,39 @@ export const VoucherDetailScreen = ({ navigation, route }: StackScreenProps<any>
     quota,
     subtitle,
     title
-  } = data[QUERY_TYPES.GENERIC_ITEM];
+  } = data?.[QUERY_TYPES.GENERIC_ITEM] ?? {};
+
+  const ids = pointOfInterest?.vouchers
+    .map((voucher: TVoucherItem) => voucher.id)
+    .filter((id: string) => id !== queryVariables?.id);
```

Comment:
please add `?` for `map` and `filter` as we do not know if `.vouchers` always return an array.

Reply:
what about using `reduce` to have a single iteration instead of two for `map` and `filter`?

```
const ids = pointOfInterest?.vouchers?.reduce((acc, voucher) => {
    if (voucher.id !== queryVariables?.id) {
        acc.push(voucher.id);
    }
    return acc;
}, []);
```

Reply:
Yeah, it's better. I'll update this.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1108/commits/19532129b36ff2bcce22bbe2a8c2e8ad90e3f93b